hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/support/validators/block/TransactionRecordParityValidator.java

```diff
@@ -0,0 +1,226 @@
+            logger.error("Validation failed. Summary: {}", validatorSummary);
+        }
+    }
+
+    private List<RecordStreamEntry> translateAll(final BlockRecordsInput inputs) {
+        // Translate the block transactions into SingleTransactionRecord instances
+        // TODO: Which state changes should be passed in here? (This is obviously wrong)
+        final var singleTxnRecs = TRANSACTION_RECORD_TRANSLATOR.translateAll(
+                inputs.txns(), inputs.allStateChanges().getFirst());
+        // Shape the translated records into RecordStreamEntry instances
+        return singleTxnRecs.stream()
+                .map(txnRecord -> {
+                    final var parts = TransactionParts.from(fromPbj(txnRecord.transaction()));
+                    final var consensusTimestamp = txnRecord.transactionRecord().consensusTimestamp();
+                    return new RecordStreamEntry(
+                            parts,
+                            pbjToProto(
+                                    txnRecord.transactionRecord(),
+                                    TransactionRecord.class,
+                                    com.hederahashgraph.api.proto.java.TransactionRecord.class),
+                            fromTimestamp(fromTimestamp(consensusTimestamp)));
+                })
+                .sorted()
+                .collect(Collectors.toList());
+    }
+
+    private List<RecordStreamEntry> transformExpectedRecords(@NonNull final RecordStreamAccess.Data data) {
+        final var numRecords = data.records().size();
+        final var txnRecs = data.records().stream()
+                .flatMap(record -> record.recordFile().getRecordStreamItemsList().stream())
+                .map(expectedTxn -> {
+                    final var consensusTimestamp = expectedTxn.getRecord().getConsensusTimestamp();
+                    return new RecordStreamEntry(
+                            TransactionParts.from(expectedTxn.getTransaction()),
+                            expectedTxn.getRecord(),
+                            fromTimestamp(consensusTimestamp));
+                })
+                .toList();
+        if (txnRecs.isEmpty()) {
+            Assertions.fail("Expected a non-empty collection of record items");
+        }
+
+        logger.info("Parsed {} record items from {} records", txnRecs.size(), numRecords);
+        return txnRecs;
+    }
+
+    private static Instant fromTimestamp(final Timestamp timestamp) {
+        return Instant.ofEpochSecond(timestamp.getSeconds(), timestamp.getNanos());
+    }
+
+    private static Timestamp fromTimestamp(final com.hedera.hapi.node.base.Timestamp timestamp) {
+        return Timestamp.newBuilder()
+                .setSeconds(timestamp.seconds())
+                .setNanos(timestamp.nanos())
+                .build();
+    }
+
+    private record BlockRecordsInput(List<SingleTransactionBlockItems> txns, List<StateChanges> allStateChanges) {}
```

Comment:
Having "Input" in the name got me confused for a moment leading me to think that it will contain only input block items.
Perhaps a name without "Input" in it might be better e.g. BlockItems?

Reply:
Agreed, "Input" in the name is confusing. I went with `BlocksData`, how's that?

Reply:
Sounds good to me, thanks!